src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureSubscriptionStep.ts

```diff
@@ -0,0 +1,158 @@
+import _ from 'lodash/fp'
+import { ReactNode, useEffect, useRef, useState } from 'react'
+import { div, h, p } from 'react-hyperscript-helpers'
+import { customSpinnerOverlay, Link, Select, useUniqueId } from 'src/components/common'
+import { ValidatedInputWithRef } from 'src/components/input'
+import { Ajax } from 'src/libs/ajax'
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData'
+import { useCancellation } from 'src/libs/react-utils'
+import { summarizeErrors } from 'src/libs/utils'
+import * as Utils from 'src/libs/utils'
+import { AzureManagedAppCoordinates } from 'src/pages/billing/models/AzureManagedAppCoordinates'
+import {
+  columnEntryStyle,
+  rowStyle
+} from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/styles'
+import { ExternalLink } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/ExternalLink'
+import { Step } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/Step'
+import {
+  LabeledField, legendDetailsStyle,
+  StepFieldLegend,
+  StepFields
+} from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepFields'
+import { StepHeader } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepHeader'
+import { validate as validateUuid } from 'uuid'
+import { validate } from 'validate.js'
+
+
+type AzureSubscriptionStepProps = {
+  isActive: boolean
+  subscriptionId?: string // undefined indicates the value hasn't been changed by the user yet
+  onSubscriptionIdChanged: (string, boolean) => void
+  managedApp?: AzureManagedAppCoordinates
+  onManagedAppSelected: (AzureManagedAppCoordinates) => void
+  inputDebounce?: number
+}
+
+const managedAppsToOptions = (apps: AzureManagedAppCoordinates[]) => _.map(application => {
+  return {
+    value: application,
+    label: !!application.region ?
+      `${application.applicationDeploymentName} (${application.region})` :
+      application.applicationDeploymentName
+  }
+}, apps)
+
+// @ts-ignore
+validate.validators.type.types.uuid = value => validateUuid(value)
+// @ts-ignore
+validate.validators.type.messages.uuid = 'must be a UUID'
+
+
+export const AzureSubscriptionStep = ({ isActive, subscriptionId, inputDebounce = 0, ...props }: AzureSubscriptionStepProps) => {
+  const [subscriptionIdError, setSubscriptionIdError] = useState<ReactNode>()
+  const [managedApps, setManagedApps] = useLoadedData<AzureManagedAppCoordinates[]>({
+    onError: state => {
+      // We can't rely on the formatting of the error, so show a generic message but include the error in the console for debugging purposes.
+      if (state.error instanceof Response) {
+        state.error.text().then(console.error)
+      } else {
+        console.error(state.error)
+      }
+      setSubscriptionIdError(h(NoManagedApps))
```

Comment:
we could display a more detailed message here, possibly derived from the response...

Reply:
We actually don't want a more detailed response. As we discussed on some recent tickets, we don't want to differentiate if the user doesn't have access to the subscription vs. if they do but have no managed apps.

We actually used to present a different message if they had access but there were no managed apps.